src/Raven.Server/RavenServerStartup.cs

```diff
@@ -44,15 +45,50 @@ public void Configure(IApplicationBuilder app, ILoggerFactory loggerfactory)
 
             _router = app.ApplicationServices.GetService<RequestRouter>();
             _server = app.ApplicationServices.GetService<RavenServer>();
+            if (IsAccessUnauthorized())
+            {
+                app.Run(UnAuthorizedRequestHandler);
+            }
             app.Run(RequestHandler);
         }
 
+        private bool IsAccessUnauthorized()
+        {
+            if (_server.Configuration.Server.AnonymousUserAccessMode == AnonymousUserAccessModeValues.None)
+                return false;
+            if (_server.Configuration.Server.AllowAdminAnonymousAccessWhenNotbindedToLocalhost == true)
+                return false;
+            var url = _server.Configuration.Core.ServerUrl.ToLowerInvariant();
+            var uri = new Uri(url);
+            //url isn't set to localhost 
+            return !(uri.IsLoopback  || uri.Host == "0.0.0.0");
+        }
+
         public static bool SkipHttpLogging;
 
+        private Task UnAuthorizedRequestHandler(HttpContext context)
+        {
+            context.Response.StatusCode = (int) HttpStatusCode.ServiceUnavailable;
+            context.Response.Headers["Content-Type"] = "application/json; charset=utf-8";
+            JsonOperationContext ctx;
+            using (_server.ServerStore.ContextPool.AllocateOperationContext(out ctx))
+            using (var writer = new BlittableJsonTextWriter(ctx, context.Response.Body))
+            {
+                writer.WriteStartObject();
+                writer.WritePropertyName("Message");
+                writer.WriteString($"The server is running in an un-authorized mode.{Environment.NewLine}" +
+                                   $"This means that Raven/AnonymousUserAccessMode is set to Admin and expose to the world.{Environment.NewLine}" +
```

Comment:
This is accurate, but you need to also explain the rational and also what is the impact of this is.

As it stands, users will read, say "oh, I need to set this config" and do that, without understanding implciations